### DISCUSSION TITLE
Github Auth Issue: Failed to fetch user profile error message

### DISCUSSION THREAD

**Discussion by uxinkc** [requarks/wiki#3581] [Error / Bug Report]
**Describe the bug**
When selecting Github as the login method, I now get "Failed to fetch user profile" error message. We have had Wiki.js installed and working for months now without any issues like this. The only change that I am aware of is adding HTTPS to our wiki.js installation on AWS.

**To Reproduce**

_Process One: Already signed in to Github_

1. Go to Login screen:

![Screen Shot 2020-07-06 at 10 56 58 PM](https://user-images.githubusercontent.com/53310223/86705980-08d35780-bfdc-11ea-9a02-f547a164bb8b.png)

2. Click on Github button to login and the page turns to this.
![Screen Shot 2020-07-06 at 10 58 03 PM](https://user-images.githubusercontent.com/53310223/86706097-286a8000-bfdc-11ea-83d9-adc5fe1b6dd3.png)

_Process Two: Not Signed in to Github_

1. Go to Login screen:

![Screen Shot 2020-07-06 at 10 56 58 PM](https://user-images.githubusercontent.com/53310223/86705980-08d35780-bfdc-11ea-9a02-f547a164bb8b.png)

2. Click on Github button to login.

3. Page loads the Sign in to Github callback screen:
![Screen Shot 2020-07-06 at 11 00 55 PM](https://user-images.githubusercontent.com/53310223/86706497-92832500-bfdc-11ea-94be-203675038117.png)

4. Redirect shows this error:
![Screen Shot 2020-07-06 at 10 58 03 PM](https://user-images.githubusercontent.com/53310223/86706097-286a8000-bfdc-11ea-83d9-adc5fe1b6dd3.png)

**Expected behavior**
Being able to sign in using Github as previously

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04, Docker
 - Wiki.js version: 2.4.107
 - Database engine: PostgreSQL 11.6

**Additional context**
As stated above, we previously have been able to login using github as our authenticator, but now we cannot.

**Comment by winkelement** [comment]
> The only change that I am aware of is adding HTTPS to our wiki.js installation on AWS.

Have you adjusted your Authorization callback URL like `https://YOUR-WIKI-DOMAIN/login/github/callback`?

**Comment by uxinkc** [comment]
@winkelement Yes, I have tried to use both `http://...` and `https://...` with the callback you listed above, but neither seem to work.

**Comment by uxinkc** [comment]
I found a solution. The Site URL in the General section wasn't set to `https`:

![Screen Shot 2020-07-07 at 4 16 03 PM](https://user-images.githubusercontent.com/53310223/86845122-a5463a00-c06e-11ea-995b-5dd6367eaecb.png)

And after I updated that to `https`, I **ALSO** had got to the Authentication section and toggle the github strategy from Active to Inactive... Click Apply... and then toggle it back to Active again and click Apply. This somehow initiated the the reset of the system to use the `https` URL as the redirect. Without doing this weird toggle action, they system would not reset to `https`

![Screen Shot 2020-07-07 at 4 46 02 PM](https://user-images.githubusercontent.com/53310223/86847053-5cdc4b80-c071-11ea-9b9e-bec0060663ab.png)

Previously, the Auth Strategy Configuration settings looks like this...

![Screen Shot 2020-07-07 at 4 16 36 PM](https://user-images.githubusercontent.com/53310223/86847711-5f8b7080-c072-11ea-9628-2fe21f78a024.png)

After updating the Site URL, the Auth Strategy Configuration settings look like the below, but did not allow for the use of Github authentication.

![Screen Shot 2020-07-07 at 4 19 28 PM](https://user-images.githubusercontent.com/53310223/86847741-729e4080-c072-11ea-9f16-a0b2950cf29e.png)

It wasn't until I performed the Toggle off/on of the github Auth and clicked Apply that the changes finally took affect.

This is for sure a bug, because a User would not expect to have to update the Site URL in a completely different section AND toggle the Authentication for github a few times after they have implemented SSL. Updating 3 different unrelated sections, seems like a lot to try and explain.

In the SSL section, if I click `TURN ON` Redirect HTTP requests to HTTPS, the system should do a system wide check for `http` and swap out with `https`.
